### PR TITLE
Handle relocatable code in AsmConstWalker

### PR DESCRIPTION
**Pull request by sbc100**
In relocatable code the constant offset might be relative to
__memory_base.

### CODE REVIEW

File: src/wasm/wasm-emscripten.cpp [hunk 1]
```diff
@@ -513,16 +513,25 @@ void AsmConstWalker::visitCall(Call* curr) {
...
     if (auto* get = arg->dynCast<GetLocal>()) {
+      // The argument may be a local.get, in which case, the last set in this
+      // basic block has the value.
       auto* set = sets[get->index];
       if (set) {
         assert(set->index == get->index);
         arg = set->value;
       }
+    } else if (auto* value = arg->dynCast<Binary>()) {
+      // If the argument is an i32.add then assume its adding a constand to the
```
Comment (dschuff) [inline src/wasm/wasm-emscripten.cpp]: typos: constand, additiona

Comment (jgravelle-google) [reply src/wasm/wasm-emscripten.cpp]: pedantry nit typo: s/its/it's/
In this case it is "it is" (it's) instead of "belonging to it" (its)

**Comment by sbc100** [comment]
Added a test case.

Updated comments.

File: src/wasm/wasm-emscripten.cpp [hunk 2]
```diff
@@ -509,20 +509,33 @@ void AsmConstWalker::visitSetLocal(SetLocal* curr) {
 
 void AsmConstWalker::visitCall(Call* curr) {
   auto* import = wasm.getFunction(curr->target);
+  // Fine calls the emscripten_asm_const* functions which first argument is
+  // alwasys a constant.
```
Comment (dschuff) [inline src/wasm/wasm-emscripten.cpp]: Not sure what "Fine calls" means (typo?). Also "alwasys"

**Pull request merged**